src/Microsoft.ML.ImageAnalytics/ImageGrayscaleTransform.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.Text;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.ImageAnalytics;
+
+[assembly: LoadableClass(ImageGrayscaleTransform.Summary, typeof(ImageGrayscaleTransform), typeof(ImageGrayscaleTransform.Arguments), typeof(SignatureDataTransform),
+    ImageGrayscaleTransform.UserName, "ImageGrayscaleTransform", "ImageGrayscale")]
+
+[assembly: LoadableClass(ImageGrayscaleTransform.Summary, typeof(ImageGrayscaleTransform), null, typeof(SignatureLoadDataTransform),
+    ImageGrayscaleTransform.UserName, ImageGrayscaleTransform.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.ImageAnalytics
+{
+    // REVIEW: Rewrite as LambdaTransform to simplify.
+    // REVIEW: Should it be separate transform or part of ImageResizerTransform?
+    /// <summary>
+    /// Transform which takes one or many columns of <see cref="ImageType"/> type in IDataView and
+    /// convert them to greyscale representation of the same image.
+    /// </summary>
+    public sealed class ImageGrayscaleTransform : OneToOneTransformBase
+    {
+        public sealed class Column : OneToOneColumn
+        {
+            public static Column Parse(string str)
+            {
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            public bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        public class Arguments : TransformInputBase
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition(s) (optional form: name:src)", ShortName = "col", SortOrder = 1)]
+            public Column[] Column;
+        }
+
+        internal const string Summary = "Convert image into grayscale.";
+
+        internal const string UserName = "Image Greyscale Transform";
+        public const string LoaderSignature = "ImageGrayscaleTransform";
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "IMGGREY ",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature);
+        }
+
+        private const string RegistrationName = "ImageGrayscale";
+
+        /// Public constructor corresponding to SignatureDataTransform.
```

Comment:
>///  [](start = 8, length = 4)

Did you mean this to be `// ` and not `/// `, since it is not really terribly sensible for callers?